go/mysql/mysql.go

```diff
@@ -52,76 +53,21 @@ const (
 	RedactedPassword = "****"
 )
 
-// SqlError is the error structure returned from calling a mysql
-// library function
-type SqlError struct {
-	Num     int
-	Message string
-	Query   string
-}
-
-// NewSqlError returns a new SqlError
-func NewSqlError(number int, format string, args ...interface{}) *SqlError {
-	return &SqlError{Num: number, Message: fmt.Sprintf(format, args...)}
-}
-
-// Error implements the error interface
-func (se *SqlError) Error() string {
-	if se.Query == "" {
-		return fmt.Sprintf("%v (errno %v)", se.Message, se.Num)
-	}
-	return fmt.Sprintf("%v (errno %v) during query: %s", se.Message, se.Num, se.Query)
-}
-
-// Number returns the internal mysql error code
-func (se *SqlError) Number() int {
-	return se.Num
-}
-
 func handleError(err *error) {
 	if x := recover(); x != nil {
-		terr := x.(*SqlError)
+		terr := x.(*sqldbconn.SqlError)
 		*err = terr
 	}
 }
 
-// ConnectionParams contains all the parameters to use to connect to mysql
-type ConnectionParams struct {
-	Host       string `json:"host"`
-	Port       int    `json:"port"`
-	Uname      string `json:"uname"`
-	Pass       string `json:"pass"`
-	DbName     string `json:"dbname"`
-	UnixSocket string `json:"unix_socket"`
-	Charset    string `json:"charset"`
-	Flags      uint64 `json:"flags"`
-
-	// the following flags are only used for 'Change Master' command
-	// for now (along with flags |= 2048 for CLIENT_SSL)
-	SslCa     string `json:"ssl_ca"`
-	SslCaPath string `json:"ssl_ca_path"`
-	SslCert   string `json:"ssl_cert"`
-	SslKey    string `json:"ssl_key"`
-}
-
-// EnableMultiStatements will set the right flag on the parameters
-func (c *ConnectionParams) EnableMultiStatements() {
-	c.Flags |= C.CLIENT_MULTI_STATEMENTS
-}
-
 // EnableSSL will set the right flag on the parameters
-func (c *ConnectionParams) EnableSSL() {
-	c.Flags |= C.CLIENT_SSL
+func EnableSSL(connParams *sqldbconn.ConnectionParams) {
```

Comment:
these are mysql specific stuff and we have to import mysql header files in order to use C.CLIENT_SSL, etc. That's why these ssl related functions still live in mysql package.